Fogl 589 - Create Payload Builder (SubTask 1 of FOGL-581)

amarendra-dianomic opened the pull request.

amarendra-dianomic commented:
Linked to [PR#201](https://github.com/foglamp/FogLAMP/pull/201)

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+        retval = False
+        if isinstance(arg, list):
+            if len(arg) == 3:
+                if arg[1] in ['<', '>', '=', '>=', '<=', 'LIKE', 'IN', '!=']:
```
**Pasting Mark's message:**
 MarkRiddoch :
 LIKE and IN are not currently supported by the storage service, these can be added if we need them

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+            self.payload.update({"limit": arg})
+        return self
+
+    def ORDERBY(self, *args):
```
ORDER_BY

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+        return self
+
+    def execute(self):
+        return dict(self.payload)
```
rename to payload, see above (constructor message)
it should return json.dumps

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+    def execute(self):
+        return dict(self.payload)
+
+    def execute_query_params(self):
```
rename to query_params (sans execute)

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+        return urllib.parse.urlencode(query_params)
+
+if __name__ == "__main__":
+    pb = PayloadBuilder(payload=OrderedDict())
```
1) better if don;t need to make it an object (should optional)
2) pb = PayloadBuilder() will work too, yes? as payload=OrderedDict() is default arg

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+        ORDERBY(['process_name', 'desc']).\
+        AGGREGATION(['count', 'process_name']).\
+        execute()
+    print(str(sql).replace("'", '"'))
```
do we need this for json.dumps too? if Yes, can we just move to execute() [to be renamed to paylaod()] itself?

praveen-garg commented on src/python/foglamp/storage/payload_builder.py:
```diff
@@ -0,0 +1,217 @@
+    print(str(sql).replace("'", '"'))
+
+    pb = PayloadBuilder(payload=OrderedDict())
+    # Insert
```
update

amarendra-dianomic commented:
@praveen-garg all changes done as advised. All changes tested successfully with examples in the code, However, pl note that:
- I doubt we can chain methods within an instance, which is very convenient feature, if we change methods to @classmethod. Because in this case, methods would return the class and NOT the instance.
- Some suggestions could not be done due to some errors for which I have added @FIXME.
- I have added a kind of very basic schema validator. We can expand on this later.
- Only task left is adding tests. Please advise @vaibhav-ScaleDB accordingly as you have suggested earlier otherwise I will take up this next week.
